plugin/loadbalance/loadbalance.go

```diff
@@ -18,9 +27,14 @@ func (r *RoundRobinResponseWriter) WriteMsg(res *dns.Msg) error {
 		return r.ResponseWriter.WriteMsg(res)
 	}
 
-	res.Answer = roundRobin(res.Answer)
-	res.Ns = roundRobin(res.Ns)
-	res.Extra = roundRobin(res.Extra)
+	switch r.policy {
+	case ramdomShufflePolicy:
+		res.Answer = roundRobin(res.Answer)
+		res.Ns = roundRobin(res.Ns)
+		res.Extra = roundRobin(res.Extra)
+	case weightedRoundRobinPolicy:
+		res.Answer = r.weights.weightedRoundRobin(res.Question[0].Name, res.Answer)
```

Comment:
I think the weighted shuffle implementation in its current state only shuffles answers to A/AAAA requests.  But if you want to shuffle responses to SRV requests too, for which any given SRV records can reference one _or more_ A/AAAA records in the `Extra` (aka Additional) section, you should also randomize Extra.  I think this would make the feature more complete - handling all (I think all) query types that can answer with A/AAAA records.

Reply:
Right, I will add code to apply the logic to the Extra section, too.

Reply:
Done.